source/Applications/Basic/Visualization/CaptureHDRVisNormals/CaptureHDRVisNormals.cpp

```diff
@@ -0,0 +1,142 @@
+        viewer->addPointCloud<pcl::PointXYZRGB>(pointCloudPCL, "cloud", view_rgb);
+
+        const int normalsSkipped = 10;
+        std::cout << "Note! 1 out of " << normalsSkipped << " normals are visualized" << std::endl;
+
+        int view_normals(0);
+        viewer->createViewPort(0.5, 0.0, 1.0, 1.0, view_normals);
+        viewer->addText("Cloud Normals", 0, 0, "NormalsText", view_normals);
+        viewer->addPointCloud<pcl::PointXYZRGBNormal>(normalsPCL, "cloudNormals", view_normals);
+        viewer->addPointCloudNormals<pcl::PointXYZRGBNormal, pcl::PointXYZRGBNormal>(normalsPCL,
+                                                                                     normalsPCL,
+                                                                                     normalsSkipped,
+                                                                                     1,
+                                                                                     "normals",
+                                                                                     view_normals);
+
+        viewer->setCameraPosition(0, 0, -100, 0, -1, 0);
+
+        std::cout << "Press r to centre and zoom the viewer so that the entire cloud is visible" << std::endl;
+        std::cout << "Press q to me exit the viewer application" << std::endl;
+        while(!viewer->wasStopped())
+        {
+            viewer->spinOnce(100);
+            std::this_thread::sleep_for(100ms);
+        }
+    }
+
+    pcl::PointCloud<pcl::PointXYZRGB>::Ptr zividToPclPoints(const Zivid::Array2D<Zivid::PointXYZColorRGBA> &data)
+    {
+        auto pointCloudPCL = boost::make_shared<pcl::PointCloud<pcl::PointXYZRGB>>();
+        pointCloudPCL->width = data.width();
+        pointCloudPCL->height = data.height();
+        pointCloudPCL->is_dense = false;
+        pointCloudPCL->points.resize(data.size());
+        for(size_t i = 0; i < data.size(); ++i)
+        {
+            pointCloudPCL->points[i].x = data(i).point.x; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL->points[i].y = data(i).point.y; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL->points[i].z = data(i).point.z; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL->points[i].r = data(i).color.r; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL->points[i].g = data(i).color.g; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL->points[i].b = data(i).color.b; // NOLINT(cppcoreguidelines-pro-type-union-access)
+        }
+        return pointCloudPCL;
+    }
+
+    pcl::PointCloud<pcl::PointXYZRGBNormal>::Ptr zividToPclVisualizationNormals(
+        const Zivid::Array2D<Zivid::PointXYZColorRGBA> &data,
+        const Zivid::Array2D<Zivid::NormalXYZ> &normals)
+    {
+        auto normalsPCL = boost::make_shared<pcl::PointCloud<pcl::PointXYZRGBNormal>>();
```

Comment:
Why do we call this normalsPCL @runenordmo?
I just noticed that it holds all data -> RGB, XYZ, and Normals.

Maybe dataWithNormalsPCL is better?

Reply:
This would now look weird :/

`visualizePointCloudAndNormalsPCL(pointCloudPCL, pointCloudWithNormalsPCL);`